### DISCUSSION TITLE
Is there a way to use model-viewer and disallow user to download the model?

### DISCUSSION THREAD

**Discussion by rafek** [google/model-viewer#4769] [Q&A] [answered]
I was wondering if there's a way to render the viewer but without exposing a link to the model being displayed? I'm looking into the source within a browser and there's a `<model-viewer>` element that has an attribute with the model link and then the canvas that renders it is somewhere deeper in the hierarchy.

Is there a way to avoid that?

**Comment by elalish** [comment]
Not really - think of a glTF as just like an image; there's nothing to stop anyone from downloading images from your site. You have legal protection with copyright, and you're welcome to hold back your top quality version and only show a downsampled / watermarked version on your website, just like photographers do. There are ways to make downloading your GLB slightly harder, but at the end of the day, it's all going to the GPU and can be extracted from there anyway.

**Reply by rafek** [reply]
Thank you @elalish for the answer. I'm wondering then how Sketchfab does that. For example https://sketchfab.com/3d-models/drive-the-wangan-c9fada2b89cd43b99a791ef59fc7bd79 - I can't find a way to download that model. Any clues?

**Reply by elalish** [reply, marked as answer]
I believe they do put some effort into obfuscating their 3D models, but those of us who debug GPUs can still access the data. Since it's not really secure (no DRM), we at MV don't feel it's worth the effort. But you can certainly use Sketchfab if you wan to make it harder for people.

**Reply by rafek** [reply]
@elalish thank you, that was most helpful! 🙏

**Comment by clochardM33** [comment]
There are a lot of words on this subject here...
#1886

**Reply by rafek** [reply]
@clochardM33 wow, that's a pretty thorough conversation. I haven't found that earlier. Thanks for referencing!

**Reply by clochardM33** [reply]
There is a big gotcha with the file swap mechanism that I've recently come across and am trying to work around.
It isn't compatible with scene-viewer AR beacuse that (seems to) need it's own independent GET to the model file back from the server.
#4790
